solarkraft/src/state/value.ts

```diff
@@ -16,7 +16,88 @@ export type Value = {
     | MapValue
 )
 
-export function isValid(v: Value): boolean {
+const EMPTY_COLLECTION_TYPE: string = 'T'
+const UNKNOWN_TYPE: string = 'UNKNOWN'
+
+// helper function
+function computeSingleTypeRep(ts: string[]): string[] {
+    if (ts.length === 0) return []
+    else {
+        const t0 = ts[0]
+        const hom = !ts.some((t) => t !== t0)
+        if (hom) return [t0]
+        else return ts
+    }
+}
+
+export function getFullType(v: Value): string {
+    switch (v.type) {
+        case 'bool':
+        case 'u32':
+        case 'i32':
+        case 'u64':
+        case 'i64':
+        case 'u128':
+        case 'i128':
+        case 'symb':
+        case 'addr':
+            return v.type
+        case 'arr':
+            return 'Bytes'
```

Comment:
How is 'arr' different from the other basic types?

Reply:
I'll add a comment about this inline, but I wanted to convey that `Bytes`, while it is a collection, is basically atomic as a type, since it's always an array of exactly 0s or 1s. I felt like using the word `array`/`arr` might give the impression that this is an array in the traditional sense, i.e. a fixed-sized ordered collection of _arbitrary_ values, which is very much not supported (but there is `Vec`).

Reply:
I have added a comment on that. I think it would improve our understanding of the issue if you added a link to the original description of this datatypes in Stellar or Soroban SDK?